src/AuthProvider.jsx

```diff
@@ -0,0 +1,36 @@
+import { createContext, useContext, useEffect, useState } from "react";
+
+const AuthContext = createContext();
+
+export const useAuth = () => {
+    return useContext(AuthContext);
+};
+
+export const AuthProvider = ({ children}) => {
+    const [isLoggedIn, setIsLoggedIn] = useState(false);
+    const [userData, setUserData] = useState({ firstName: "", lastName: "", role: "" })
+    const isToken = sessionStorage.getItem('isToken');
```

Comment:
You are accessing session storage synchronously. This could lead to some performance issues. It's better to initialize this function in `isLoggedIn` state asynchronously.